packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -772,6 +813,89 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
+            if (parent && parent.children.length > 0) {
+                for (const sibling of parent.children) {
+                    this.nodesWithActiveIndentGuideline.add(sibling.id);
+                    if (this.isExpandable(sibling) && sibling.expanded && sibling.children.length > 0) {
+                        for (const child of sibling.children) {
+                            for (const grandchild of new DepthFirstTreeIterator(child)) {
+                                this.siblingChildNodesWithActiveIndentGuideline.add(grandchild.id);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Render indent for the file tree depending on the depth
+     * @param node the tree node.
+     * @param depth the depth of the tree node
+     */
+    protected renderIndent(node: TreeNode, depth: number): React.ReactNode {
+        if (this.nodesWithActiveIndentGuideline.has(node.id)) {
+            if (this.minIndentGuidelineDepth === 0) {
+                this.minIndentGuidelineDepth = depth;
+            } else {
+                if (depth < this.minIndentGuidelineDepth) {
+                    this.minIndentGuidelineDepth = depth;
+                }
+            }
+        }
+
+        if (this.nodesWithActiveIndentGuideline.has(node.id)) {
+            if (this.maxIndentGuidelineDepth === 0) {
+                this.maxIndentGuidelineDepth = depth;
+            } else {
+                if (depth > this.maxIndentGuidelineDepth) {
+                    this.maxIndentGuidelineDepth = depth;
+                }
+            }
+        }
+
+        const indentDivs: React.ReactNode[] = [];
+        const needsNodeGuideline = this.nodesWithActiveIndentGuideline.has(node.id);
+        const needsSiblingGuideline = this.siblingChildNodesWithActiveIndentGuideline.has(node.id);
+        const minDepthLine = this.minIndentGuidelineDepth;
+        const maxDepthLine = this.maxIndentGuidelineDepth;
+        const indentGuideOption = this.corePreferences['workbench.tree.renderIndentGuides'];
+        for (let i = 0; i < depth; i++) {
+            const needsLeafPadding = (!this.isExpandable(node) && i === (depth - 1));
+            indentDivs.push(<div key={i}
+                className={`${TREE_NODE_INDENT_BLOCK_CLASS} 
+                ${indentGuideOption !== 'none' ? indentGuideOption === 'onHover' ? INDENT_GUIDE_ONHOVER : INDENT_GUIDE_ALWAYS : ''}  
+                ${needsLeafPadding ? TREE_NODE_CHILD_PADDING_CLASS : ''}
+                ${indentGuideOption !== 'none' && (needsNodeGuideline || needsSiblingGuideline)
+                        && ((i === minDepthLine - 1) || (needsNodeGuideline && i === (depth - 1))
+                            || (this.numOfNodesSelected > 1 && needsSiblingGuideline
+                                && i > (minDepthLine - 1) && i === (maxDepthLine - 1))) ?
+                        TREE_NODE_ACTIVE : ``}
```

Comment:
@kaiyue0329 use single quotes here ` `` ` -> `''` (reason for the [CI failure](https://travis-ci.com/eclipse-theia/theia/jobs/289908374#L846))